File: etsyCatalogs/etsyCatalogs/Model/CatalogResult.m
//
//  CatalogResult.m
//  etsyCatalogs
//
//  Created by Ihonahan Buitrago on 9/2/14.
//  Copyright (c) 2014 Ihonahan Buitrago. All rights reserved.
//

#import "CatalogResult.h"


@interface CatalogResult()

@property(nonatomic, strong) NSMutableArray *listingResults;

@end


@implementation CatalogResult


- (id)init
{
    self = [super init];
    if (self) {
        // We need to init the array to be sure it will be there when we need it
        self.listingResults = [NSMutableArray array];
    }
    
    return self;
}


- (void)insertPaginationInfo:(NSDictionary *)dictionary withCount:(int)count
{
    self.pagination = [[ResultPagination alloc] init];
    self.pagination.pageLimit = [[dictionary objectForKey:@"effective_limit"] intValue];
    self.pagination.currentOffset = [[dictionary objectForKey:@"effective_offset"] intValue];
    self.pagination.nextOffset = [[dictionary objectForKey:@"next_offset"] intValue];
    self.pagination.currentPage = [[dictionary objectForKey:@"effective_page"] intValue];
    self.pagination.nextPage = [[dictionary objectForKey:@"next_page"] intValue];
    self.pagination.totalCount = count;
}

- (void)insertCatalogItem:(NSDictionary *)dictionary
{
    CatalogItem *item = [CatalogItem createCatalogItemFromDictionary:dictionary];

    [self.listingResults addObject:item];
}

- (CatalogItem *)getCatalogItemWithListingID:(long long)listingId
{
    for (CatalogItem *item in self.listingResults) {
        if (item.listingID == listingId) {
            return item;
        }
    }
    
    return nil;
}

- (CatalogItem *)getCatalogItemByIndex:(int)index
{
    if (index < self.listingResults.count) {
        return [self.listingResults objectAtIndex:index];
    } else {
        return nil;
    }
}

- (NSMutableArray *)getAllCatalogItemsCopy
{
    return [self.listingResults copy];
}

- (int)count
{
    return self.listingResults.count;
}

@end

File: etsyCatalogsSwift/etsyCatalogsSwift/Model/CatalogResult.swift
//
//  CatalogResult.swift
//  etsyCatalogsSwift
//
//  Created by Ihonahan Buitrago on 9/22/14.
//  Copyright (c) 2014 Ihonahan Buitrago. All rights reserved.
//

import Foundation

class CatalogResult : NSObject
{
    var pagination : ResultPagination = ResultPagination()
    var listingResults : [CatalogItem] = []

    
    override init()
    {
        super.init()
    }

    
    func insertPaginationInfo(dictionary : Dictionary<String, AnyObject>, count : Int)
    {
        var pageDictionary : Dictionary<String, AnyObject> = dictionary
        pageDictionary.updateValue(count, forKey: "count")
        self.pagination = ResultPagination.createCatalogListingImageFromDictionary(pageDictionary)
    }
    
    
    func insertCatalogItem(dictionary : Dictionary<String, AnyObject>)
    {
        var item : CatalogItem = CatalogItem.createCatalogItemFromDictionary(dictionary)
        
        self.listingResults.append(item)
    }
    
    
    func getCatalogItemWithListingID(listingId : Int) -> CatalogItem?
    {
        var idList : Int? = 0
        for item: CatalogItem in self.listingResults {
            idList = item.listingID;
            
            if idList == listingId {
                return item
            }
        }
        
        return nil
    }
    
    func getCatalogItemByIndex(index : Int) -> CatalogItem?
    {
        if index < self.listingResults.count {
            return self.listingResults[index]
        } else {
            return nil
        }
    }
    
    
    func getAllCatalogs() -> Array<CatalogItem>
    {
        return self.listingResults
    }
    
    
    func count() -> Int
    {
        return self.listingResults.count
    }
    
    
    func findIndexOfItem(item : CatalogItem) -> Int
    {
        var i = 0
        for listItem in self.listingResults {
            if listItem == item {
                return i
            }
            i++;
        }
        
        return -1
    }
}


